examples/ASDKgram/Sample/CommentsNode.m

```diff
@@ -0,0 +1,101 @@
+//
+//  CommentsNode.m
+//  ASDKgram
+//
+//  Created by Hannah Troisi on 3/21/16.
+//  Copyright © 2016 Hannah Troisi. All rights reserved.
+//
+
+#import "CommentsNode.h"
+
+#define INTER_COMMENT_SPACING 5
+#define NUM_COMMENTS_TO_SHOW 3
+
+@implementation CommentsNode
+{
+  CommentFeedModel              *_commentFeed;
+  NSMutableArray <ASTextNode *> *_commentNodes;
+}
+
+#pragma mark - Lifecycle
+
+- (instancetype)init
+{
+  self = [super init];
+  if (self) {
+    _commentNodes = [[NSMutableArray alloc] init];
+  }
+  return self;
+}
+
+- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
+{
+  ASStackLayoutSpec *verticalStack = [ASStackLayoutSpec verticalStackLayoutSpec];
+  verticalStack.spacing            = INTER_COMMENT_SPACING;
+  [verticalStack setChildren:_commentNodes];
+  
+  return verticalStack;
+}
+
+#pragma mark - Instance Methods
+
+- (void)updateWithCommentFeedModel:(CommentFeedModel *)feed
+{
+  _commentFeed = feed;
+  [self removeCommentLabels];
+  
+  if (_commentFeed) {
+    [self createCommentLabels];
+    
+    BOOL addViewAllCommentsLabel = [feed numberOfCommentsForPhotoExceedsInteger:NUM_COMMENTS_TO_SHOW];
+    NSAttributedString *commentLabelString;
+    int labelsIndex = 0;
+    
+    if (addViewAllCommentsLabel) {
+      commentLabelString        = [_commentFeed viewAllCommentsAttributedString];
+      [[_commentNodes objectAtIndex:labelsIndex] setAttributedString:commentLabelString];
+      labelsIndex++;
+    }
+    
+    NSUInteger numCommentsInFeed = [_commentFeed numberOfItemsInFeed];
+    
+    for (int feedIndex = 0; feedIndex < numCommentsInFeed; feedIndex++) {
+      commentLabelString         = [[_commentFeed objectAtIndex:feedIndex] commentAttributedString];
+      [[_commentNodes objectAtIndex:labelsIndex] setAttributedString:commentLabelString];
+      labelsIndex++;
+    }
+    
+    [self setNeedsLayout];
+  }
+}
+
+
+#pragma mark - Helper Methods
+
+- (void)removeCommentLabels
+{
+  for (ASTextNode *commentLabel in _commentNodes) {
+    [commentLabel removeFromSupernode];
```

Comment:
You should refactor this code - using implicit hierarchy management will be much simpler.  No need for any of the add / remove as far as I know, since no reuse of these items as well.